lib/forms/CoBrALab-Mouse-Injection-Form/index.ts

```diff
@@ -0,0 +1,228 @@
+          return {
+            kind: "string",
+            variant: "select",
+            label: "Analgesic type",
+            options: {
+              "Carpofem": "Carpofem",
+              "Buvicane": "Buvicane"
+            }
+          }
+        }
+        return null
+      }
+    },
+
+    ipDoseVolume: createDependentField({
+      kind: "number",
+      variant: "input",
+      label: "IP dose volume"
+    }, (type) => type === "IP"),
+
+    drugInjected: createDependentField({
+      kind: "string",
+      variant: "select",
+      label: "Drug injected",
+      options: {
+        "PU-AD":"PU-AD",
+        "PU-AD Vehicle": "PU-AD Vehicle",
+        "IP Tamoxifen": "IP Tamoxifen",
+        "STZ": "STZ"
+      }
+    }, (type) => type === "IP"),
+
+    ipInjectionType: createDependentField({
+      kind: "string",
+      variant: "select",
+      label: "IP injection type",
+      options: {
+        "Viral memetic": "Viral memetic",
+        "Anesthetic": "Anesthetic"
+      }
+    },(type) => type === "IP")
+
+    
+  },
+  details: {
+    description: 'A form to describe a mouses injection information',
+    estimatedDuration: 1,
+    instructions: ['Please fill out this form whenever a mouse experiences an injection'],
+    license: 'UNLICENSED',
+    title: 'Injections Form'
+  },
+  measures: {
+    roomNumber: {
+      kind: "const",
+      ref: "roomNumber"
+    },
+    injectionType: {
+      kind: "const",
+      ref: "injectionType"
+    },
+    intracerebralInjectionType: {
+      kind: "const",
+      ref: "intracerebralInjectionType"
+    },
+    hydrationProvided: {
+      kind: "const",
+      ref: "hydrationProvided"
+    },
+    hydrationVolume: {
+      kind: "const",
+      ref: "hydrationVolume"
+    },
+    subcutaneousInjectionType: {
+      kind: "const",
+      ref: "subcutaneousInjectionType"
+    },
+    subcutaneousInjectionTime: {
+      kind: "const",
+      ref: "subcutaneousInjectionTime"
+    },
+    postOperationDay: {
+      kind: "const",
+      ref: "postOperationDay"
+    },
+    analgesicType: {
+      kind: "const",
+      ref: "analgesicType"
+    },
+    ipDoseVolume: {
+      kind: "const",
+      ref: "ipDoseVolume"
+    },
+    drugInjected: {
+      kind: "const",
+      ref: "drugInjected"
+    },
+    ipInjectionType: {
+      kind: "const",
+      ref: "ipInjectionType"
+    }
+
+  },
+  validationSchema: z.object({
+    roomNumber: z.string(),
+    injectionType: z.string(),
+    intracerebralInjectionType: z.string().optional(),
+    hydrationProvided: z.boolean().optional(),
+    hydrationVolume: z.string().optional(),
+    subcutaneousInjectionType: z.string().optional(),
+    subcutaneousInjectionTime: z.string().optional(),
+    postOperationDay: z.date().optional(),
+    analgesicType: z.string().optional(),
+    ipDoseVolume: z.number().optional(),
+    drugInjected: z.string().optional(),
+    ipInjectionType: z.string().optional()
+  })
```

Comment:
should volumes be optional?

Reply:
volumes are optional since they're only shown for certain injection types (i.e. saline hydration for intracerebral injection, etc.). Without the optional tag the form would not be able to submit if these questions are unanswered, even when they questions are hidden.

Reply:
:+1: